### ISSUE TITLE
Plugin can't be pushed via WP Engine due to their linting pre-push

### ISSUE THREAD

**Issue by jiwicky** [CalderaWP/Caldera-Forms#834]
Hi,
WP Engine's git push does a linting of all php files before allowing a push to happen. 

A composer file (/classes/email/sendgrid/vendor/composer/autoload_static.php) is causing the linting to fail and therefore not allowing a push to WP Engine due to reasons outlined here:
https://github.com/composer/composer/issues/5324

(more discussion here: https://github.com/composer/composer/pull/5660)

I know it can be argued that WP Engine should be running a newer version of PHP but I wanted to bring this to your attention anyway. 

Thanks!

**Comment by Shelob9** [comment]
Thanks for bringing this to my attention. I'm going to look into this with WPEngine. Linting PHP is a problem as it doesn't take into account the backwards compatibility checks. Composer loads a specific autoloader for each version of PHP so it can use the best syntax without breaking anything.

**Comment by Shelob9** [comment]
@jiwicky I'm going to close this, because it's a conflict, not a bug and I'm trying to keep things clear here. Please feel free to keep commenting for follow up. I will let you know what I hear back from WPEngine.

**Comment by jiwicky** [comment]
ok, thanks @Shelob9

**Comment by anthonyburchell** [comment]
Hey @jiwicky !
I'm am with our Support Operations team here at WP Engine. @Shelob9 reached out to me and got this issue on my radar. I have followed up with our team responsible for the platform and we are already taking the steps to finding a better process. Linting these days is far from perfect and we're trying to find the right way to get it there. 

We currently do not have a solid workaround, but this issue has been known and ongoing to some degree. We did not know the scope of impact included composer so this is good information to have. I am relaying this ticket to our engineers to decide how we should move forward.

I'm sure this is frustrating in your workflow so I appreciate your patience as we work through it. If I find any workaround or have any info to follow up on I will comment back in this issue.

**Comment by jiwicky** [comment]
Thanks @anthonyburchell ! Looking forward to the improved approach.

**Comment by impartdan** [comment]
+1

**Comment by thowar2** [comment]
If anyone is looking for a quick fix for this issue, I've added `composer/autoload_static.php` to .gitignore so I can deploy to WP Engine without having the repo break WP Engine every time someone sets up composer locally.

**Comment by anthonyburchell** [comment]
Hi there @impartdan  and @jiwicky ! 

@thowar2 Thank you for providing that workaround for everyone. Really appreciate that. 

Just wanted you all to get the update that we have rolled out changes that will no longer factor autoload_static.php in the linting process. As @thowar2 suggested the way this should be done is by adding the file location to .gitignore but for the sake of saving the work in the future we've just excluded it from the process altogether. 

Thanks for the patience in this!
